Common/src/Auth/Authorization/Permissions.cs

```diff
@@ -0,0 +1,115 @@
+﻿// This file is part of the ArmoniK project
+// 
+// Copyright (C) ANEO, 2021-2022. All rights reserved.
+//   W. Kirschenmann   <wkirschenmann@aneo.fr>
+//   J. Gurhem         <jgurhem@aneo.fr>
+//   D. Dubuc          <ddubuc@aneo.fr>
+//   L. Ziane Khodja   <lzianekhodja@aneo.fr>
+//   F. Lemaitre       <flemaitre@aneo.fr>
+//   S. Djebbar        <sdjebbar@aneo.fr>
+//   J. Fonseca        <jfonseca@aneo.fr>
+//   D. Brasseur       <dbrasseur@aneo.fr>
+// 
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published
+// by the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+// 
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+// 
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Security.Claims;
+
+using ArmoniK.Core.Common.gRPC.Services;
+
+namespace ArmoniK.Core.Common.Auth.Authorization
+{
+  public static class Permissions
+  {
+    public class Permission
+    {
+      public readonly string  Prefix;
+      public readonly string  Name;
+      public readonly string? Suffix;
+      public readonly Claim   Claim;
+
+      public Permission(string actionString)
+      {
+        var parts = actionString.Split(Separator);
+        if (parts.Length is < 2 or > 3)
+        {
+          throw new ArgumentOutOfRangeException("Wrong number of parts in action policy string " + actionString);
+        }
+
```

Comment:
Couldn't we use use pattern matching here: https://docs.microsoft.com/en-us/dotnet/csharp/fundamentals/functional/pattern-matching

Reply:
no regex, no. You can keep it like this.